src/main/java/carly/Carly.java

```diff
@@ -0,0 +1,137 @@
+     * appropriate exceptions are caught, and error messages are displayed without terminating the chat.
+     */
+    private void chat() {
+        Scanner scan = new Scanner(System.in);
+        Ui ui = new Ui();
+        ui = ui.setUsername(scan);
+        ui.welcomeMsg();
+
+        String input;
+        String taskDescription;
+        Parser.Command command;
+        Storage listStorage = new Storage("./data/CarlyList.txt");
+
+        while (true) {
+            try {
+                input = ui.readInput(scan);
+                Parser parser = new Parser(input);
+                command = parser.getCommand();
+                taskDescription = parser.getDetailsAfterCommand(command);
+            } catch (CarlyException e) {
+                System.out.println(e.getMessage());
+                continue;
+            }
+
+            switch (command) {
+            case BYE:
+                ui.byeMsg();
+                return;
+            case LIST:
+                this.taskList.printTaskList();
+                break;
+            case MARK:
+                try {
+                    this.taskList.mark(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case UNMARK:
+                try {
+                    this.taskList.unmark(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case DELETE:
+                try {
+                    this.taskList.delete(taskDescription);
+                    break;
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case FIND:
+                try {
+                    this.taskList.find(taskDescription);
+                    break;
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case TODO:
+                try {
+                    this.taskList.addToDo(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case DEADLINE:
+                try {
+                    this.taskList.addDeadLine(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            case EVENT:
+                try {
+                    this.taskList.addEvent(taskDescription);
+                } catch (CarlyException e) {
+                    System.out.println(e.getMessage());
+                }
+                break;
+            default:
+                Ui.printOutput("Oops, please print a valid command! ");
```

Comment:
Perhaps I'm reading this incorrectly, but this seems like the error message for an invalid command? In which case, your `CarlyUnknownIInputException` should've handled it?